### ISSUE TITLE
Actions: support setting a service response as attachment

### ISSUE THREAD

**Issue by ebruchez** [orbeon/orbeon-forms#3828]
In addition to "Set Control Value", we should have a "Set Attachment Value". This would:

- read the body as binary
- store it as a temp file as if it was a file upload
- P2: support "accept" mediatypes/size like a client upload, maybe going through the same filter when reading

[+1 from customer](https://3.basecamp.com/3600924/buckets/7873395/messages/1433223067#__recording_1441330796)

See also #3697.

**Comment by ebruchez** [comment]
- [ ] tell customer, when done, to remove hack of calling `NetUtils` function if possible

**Comment by ebruchez** [comment]
One question is whether this should be part of the legacy UI or just part of the 2018.2 action syntax for now (adding the UI later as we plan to, see #1298). I think that we can go for the latter and add:

```xml
<fr:control-setfile
    control="my-attachment"/>
```

or:

```xml
<fr:control-setattachment
    control="my-attachment"/>
```

Now how do we pass the body returned from a service to the control? `fr:service-call` just `xf:send`s the service submission. What happens if the response body is binary content? We need a way to handle that as a response, but currently we don't, I believe!

**Comment by ebruchez** [comment]
For customer the use case is ok having the returned data originally in `base64Binary` format.

**Comment by ebruchez** [comment]
After further discussion, attachments are more likely to be returned in binary form. So we should handle that for sure.

**Comment by ebruchez** [comment]
If the attachment comes back in binary form, and if the filename/mediatype are to be set, then we would need a way to access those values, maybe via expressions. They might come from HTTP headers, typically.

The new services set an implicit context containing the response document in XML. If we have a binary response, what would that contain? We also discussed the ability for a service call to assign to a variable a response object, which would give access to more aspects of the response, including of course headers. So we might need to do all of that to implement this properly.

**Comment by ebruchez** [comment]
Entered #3945 for binary handling.

**Comment by ebruchez** [comment]
How will #3945 work in this context? `replace="xxf:binary"` would have to be placed on the *service*.

- Declaring a service as returning "binary" is not entirely ridiculous. It is probably rare that you would declare a service as returning XML or JSON, for use in a dataset, for example, but then also want that response to be stored as an attachment.
- In the big picture, it *should* be possible to do just that, but it's probably not a priority.
- One could argue that to do that you must read the response body twice, once to store it as is and once to parse it, and that it's getting complicated.
- In short: probably ok to have this limitation for now.

The service response would be stored into the service response instance as usual, possibly with content-type and other service response information (or that might stay in the done event).

The only subsequent meaningful action, for now, would be "set service response as attachment".

Which actions make sense in the context of which service call would be, as a first step, left to the author of the actions. Later, the UI and/or runtime could check.

**Comment by ebruchez** [comment]
Sample form which loads an image: https://gist.github.com/ebruchez/b09399d44b64896608f0d380eb5b0340
